internal/cloud/cloud_discovery.go

```diff
@@ -47,7 +49,12 @@ func NewDiscoveryClient(config *cluster.CloudConfig, logger logger.LogAdaptor) *
 func (c *DiscoveryClient) DiscoverNodes(ctx context.Context) ([]Address, error) {
 	url := makeCoordinatorURL(c.token)
 	if j, err := c.httpClient.GetJSONArray(ctx, url); err != nil {
-		return nil, err
+		switch err.(type) {
+		case *url2.Error:
+			return nil, hzerrors.NewInvalidConfigurationError(fmt.Sprintf("failed to reach Hazelcast Cloud/Viridian service: %s", baseURL()), nil)
+		default:
+			return nil, err
+		}
```

Comment:
Instead of checking the error here, how about doing that in `HTTPClient.Get` around line 68:
```
if resp, err := c.httpClient.Do(req); err != nil {
	var e *url.Error
	if errors.As(err, &e) {
		e.URL = ""
	}
	return nil, err
// ...
```
In order to use `url.Error`, you will have to change the `url` parameter to something like `uri`.

Reply:
IMO, removing the URL from error is not convenient since HTTPClient is more general purpose. But I searched and couldn't find any other usage of it so moved the check to there. See https://github.com/hazelcast/hazelcast-go-client/pull/905/commits/89df180ed0646b11b9cd31e1e675ffa78d8c5794.